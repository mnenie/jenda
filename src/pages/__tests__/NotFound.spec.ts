import { shallowMount } from '@vue/test-utils';
import { describe, it, expect, vi } from 'vitest';
import NotFound from '../NotFoundPage.vue';
import { useHead } from '@unhead/vue';

vi.mock('@unhead/vue', () => ({
  useHead: vi.fn()
}));

const mockRouter = {
  push: vi.fn(),
  beforeEach: vi.fn()
};

describe('tests for NotFoundPage.vue', () => {
  const wrapper = shallowMount(NotFound, {
    global: {
      mocks: {
        $router: mockRouter
      }
    }
  });

  it('should be render correctly', () => {
    expect(wrapper.html());
  });

  it('should has correct head', () => {
    expect(useHead).toHaveBeenCalledWith({
      title: '404 | Not Found :/'
    });
  });

  it('should redirect correctly', async () => {
    await wrapper.findComponent({ name: 'UiButton' }).trigger('click');
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'boards' });
  });
});
