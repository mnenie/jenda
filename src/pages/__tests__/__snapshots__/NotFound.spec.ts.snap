// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for NotFoundPage.vue > should be render correctly 1`] = `
"<div class="container">
  <div class="items">
    <h1>404</h1>
    <h3>Not Found</h3>
    <p>The page you were looking for can't be found</p>
    <ui-button-stub variant="default" size="md"></ui-button-stub>
  </div>
</div>"
`;
